website_mrp_bom/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteMrpBomPage(http.Controller):
+
+    @http.route(['/components/<model("mrp.bom"):bom>'], type='http',
+                auth="public", website=True)
+    def boms_detail(self, bom, **post):
+        if isinstance(bom, request.env['mrp.bom'].__class__):
+            if bom.website_published or self._is_website_publisher(request):
+                values = {
+                    'main_object': bom,
+                    'bom': bom,
+                    'has_published_bom': self._has_published_bom(bom)
```

Comment:
seems redundant. _has_published_bom could be called using the above. being a staticmethod you would have to write bom._has_published_bom(bom), wich is ugly. But i think you can remove some redundancy here.

Reply:
I don't get you